js/data/bucket.js

```diff
@@ -76,10 +77,33 @@ function Bucket(options) {
 
     if (options.elementGroups) {
         this.elementGroups = options.elementGroups;
-        this.buffers = util.mapObject(options.arrays, function(array, bufferName) {
-            var arrayType = options.arrayTypes[bufferName];
-            var type = (arrayType.members.length && arrayType.members[0].name === 'vertices' ? Buffer.BufferType.ELEMENT : Buffer.BufferType.VERTEX);
-            return new Buffer(array, arrayType, type);
+
+        for (var programName in this.elementGroups) {
+            var elementGroups = this.elementGroups[programName];
+            if (elementGroups) {
```

Comment:
I'm not thrilled with the idea of using `if (elementGroups)` as a way to check if this is running on the main thread or in the worker. However, I appreciate why this is really hard given the current architecture of the `Bucket` class. Whenever you are done making changes to this part of the system, I would like to refactor `Bucket` 👍